patroni/config_generator.py

```diff
@@ -0,0 +1,274 @@
+                     AND category <> 'Write-Ahead Log / Recovery Target' \
+                     AND setting <> '(disabled)' \
+                     OR name IN ('hba_file', 'ident_file', 'config_file', \
+                                 'data_directory', \
+                                 'listen_addresses', 'port', \
+                                 'max_connections', 'max_worker_processes', 'max_wal_senders', \
+                                 'max_replication_slots', 'max_locks_per_transaction', 'max_prepared_transactions', \
+                                 'hot_standby', \
+                                 'wal_level', 'wal_log_hints', \
+                                 'wal_keep_segments', 'wal_keep_size', \
+                                 'track_commit_timestamp');")
+
+        helper_dict = dict.fromkeys(['port', 'listen_addresses'])
+        # adjust values
+        for p, v in cur.fetchall():
+            if p == 'data_directory':
+                config['postgresql']['data_dir'] = v
+            elif p == 'cluster_name':
+                config['scope'] = v
+            elif p in ('archive_command', 'restore_command', 'archive_cleanup_command',
+                       'recovery_end_command', 'ssl_passphrase_command',
+                       'hba_file', 'ident_file', 'config_file'):
+                # write commands to the local config due to security implications
+                # write hba/ident/config_file to local config to ensure they are not removed later
+                config['postgresql'].setdefault('parameters', {})
+                config['postgresql']['parameters'][p] = v
+            elif p in ('port', 'listen_addresses'):
+                helper_dict[p] = v
+            else:
+                config['bootstrap']['dcs']['postgresql']['parameters'][p] = v
+
+    conn.close()
+
+    connect_port = parsed_dsn.get('port', os.getenv('PGPORT', helper_dict['port']))
+    config['postgresql']['connect_address'] = f'{get_ip()}:{connect_port}'
+    config['postgresql']['listen'] = f'{helper_dict["listen_addresses"]}:{helper_dict["port"]}'
+
+    # it makes sense to define postgresql.pg_hba/pg_ident only if hba_file/ident_file are set to defaults
+    default_hba_path = os.path.join(config['postgresql']['data_dir'], 'pg_hba.conf')
+    if config['postgresql']['parameters']['hba_file'] == default_hba_path:
+        try:
+            with open(default_hba_path, 'r') as f:
+                config['postgresql']['pg_hba'] = [i.strip() for i in f.readlines()
+                                                  if i.startswith(('local',
+                                                                   'host',
+                                                                   'hostssl',
+                                                                   'hostnossl',
+                                                                   'hostgssenc',
+                                                                   'hostnogssenc'))]
```

Comment:
```
localbla
```
the code will happily consume this line.

We have to check that there is a space or tab after the approved word.
The easiest way of doing it: `i.split()[0] in ('local', ...)`

Reply:
however this approach doesn't cover empty lines in the file

Reply:
```
if i.strip().startswith(('local ', 'host ', ...))
```
is this too ugly?